### PR TITLE
[SER-474] Flatbuffer codegen uses cargo build script

### PR DESCRIPTION
**Pull request by TheButlah**
This PR required me to fork and heavily modify the dynpath crate. I hope that my upstream contribution will be merged, if not we will just keep using my fork or move my fork into 0xTELEPORTAL.

https://github.com/danakj/dynpath/pull/2

### CODE REVIEW

File: client/serialize/rust/build.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+use std::path::{Path, PathBuf};
+
+fn main() -> flatc_rust::Result<()> {
+    println!("cargo:rerun-if-changed=../flatbuffers");
+
+    flatc_rust::run(flatc_rust::Args {
+        inputs: &[&Path::new("../flatbuffers/all.fbs")],
+        // out_dir: &Path::new("src/generated"),
```
Comment (tomthecarrot) [inline client/serialize/rust/build.rs]: Still necessary?

**Comment by tomthecarrot** [comment]
> This PR required me to fork and heavily modify the dynpath crate.
> 
> https://github.com/danakj/dynpath/pull/2

Wow that is quite the modification :o

File: crates/rsharp/cs/test/RBox.cs [hunk 2]
```diff
@@ -20,7 +20,7 @@ public void TestU8()
         {
             output.WriteLine("Starting RBox U8 test");
             var u8 = new RBox_U8(10);
-            Assert.Equal(10, u8.Value);
+            ssert.Equal(10, u8.Value);
```
Comment (tomthecarrot) [inline crates/rsharp/cs/test/RBox.cs]: Typo?

Comment (TheButlah) [reply crates/rsharp/cs/test/RBox.cs]: oops, fixed

File: client/serialize/rust/build.rs (hunk 1 above)
Comment (TheButlah) [reply client/serialize/rust/build.rs]: Fixed

**Review by tomthecarrot — APPROVED** [review]

**Pull request merged**